index.js

```diff
@@ -143,6 +143,11 @@ function filter(patterns, opts) {
   }
 
   patterns = utils.arrayify(patterns);
+  var len = patterns.length, i = 0;
+  while (i < len) {
+    patterns[i] = matcher(patterns[i++], opts);
```

Comment:
You overwrote the user input of patterns. Would just like to see it explicitly reflected that now the array is something different (matcher functions).